pkg/controller/jobs/pod/pod_controller_test.go

```diff
@@ -0,0 +1,489 @@
+package pod
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	rayjobapi "github.com/ray-project/kuberay/ray-operator/apis/ray/v1alpha1"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/utils/ptr"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	kueue "sigs.k8s.io/kueue/apis/kueue/v1beta1"
+	controllerconsts "sigs.k8s.io/kueue/pkg/controller/constants"
+	"sigs.k8s.io/kueue/pkg/controller/jobframework"
+	_ "sigs.k8s.io/kueue/pkg/controller/jobs/job"
+	utiltesting "sigs.k8s.io/kueue/pkg/util/testing"
+	testingutil "sigs.k8s.io/kueue/pkg/util/testingjobs/pod"
+)
+
+func TestPodsReady(t *testing.T) {
+	testCases := map[string]struct {
+		pod  *corev1.Pod
+		want bool
+	}{
+		"pod is ready": {
+			pod: testingutil.MakePod("test-pod", "test-ns").
+				Queue("test-queue").
+				StatusConditions(
+					corev1.PodCondition{
+						Type:   "Ready",
+						Status: "True",
+					},
+				).
+				Obj(),
+			want: true,
+		},
+		"pod is not ready": {
+			pod: testingutil.MakePod("test-pod", "test-ns").
+				Queue("test-queue").
+				StatusConditions().
+				Obj(),
+			want: false,
+		},
+	}
+
+	for name, tc := range testCases {
+		t.Run(name, func(t *testing.T) {
+			pod := (*Pod)(tc.pod)
+			got := pod.PodsReady()
+			if tc.want != got {
+				t.Errorf("Unexpected response (want: %v, got: %v)", tc.want, got)
+			}
+		})
+	}
+}
+
+func TestRunWithPodSetsInfo(t *testing.T) {
+	testCases := map[string]struct {
+		pod                  *Pod
+		runInfo, restoreInfo []jobframework.PodSetInfo
+		wantPod              *corev1.Pod
+		wantErr              error
+	}{
+		"pod set info > 1": {
+			pod:     (*Pod)(testingutil.MakePod("test-pod", "test-namespace").Obj()),
+			runInfo: make([]jobframework.PodSetInfo, 2),
+			wantErr: fmt.Errorf("invalid podset infos: expecting 1 got 2"),
```

Comment:
```suggestion
			wantErr: jobframework.ErrInvalidPodsetInfo,
```

We should compare wrapped errors instead of comparing error messages.